OrbitQt/ConnectToStadiaWidget.cpp

```diff
@@ -0,0 +1,344 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ConnectToStadiaWidget.h"
+
+#include <QMessageBox>
+#include <QRadioButton>
+#include <QSettings>
+#include <optional>
+
+#include "Error.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/Result.h"
+#include "OrbitGgp/Client.h"
+#include "OverlayWidget.h"
+#include "servicedeploymanager.h"
+
+namespace {
+const QString kRememberChosenInstance{"RememberChosenInstance"};
+}  // namespace
+
+namespace orbit_qt {
+
+using OrbitGgp::Instance;
+using OrbitSshQt::ScopedConnection;
+
+ConnectToStadiaWidget::ConnectToStadiaWidget(QWidget* parent)
+    : QWidget(parent),
+      ui_(std::make_unique<Ui::ConnectToStadiaWidget>()),
+      settings_("The Orbit Authors", "Orbit Profiler") {
+  ui_->setupUi(this);
+
+  if (!settings_.value(kRememberChosenInstance).toString().isEmpty()) {
+    remembered_instance_id_ = settings_.value(kRememberChosenInstance).toString();
+    ui_->rememberCheckBox->setChecked(true);
+  }
+
+  ui_->instancesTableView->setModel(&instance_model_);
+
+  QObject::connect(ui_->connectToStadiaInstanceRadioButton, &QRadioButton::toggled, this,
+                   [this](bool checked) {
+                     if (checked) {
+                       emit Selected();
+                     } else {
+                       ui_->connectToStadiaInstanceRadioButton->setChecked(true);
+                     }
+                   });
+
+  QObject::connect(this, &ConnectToStadiaWidget::ErrorOccurred, this,
+                   [this](const QString& message) {
+                     QMessageBox::critical(this, QApplication::applicationName(), message);
+                   });
+
+  QObject::connect(ui_->instancesTableView->selectionModel(), &QItemSelectionModel::currentChanged,
+                   this, [this](const QModelIndex& current) {
+                     if (!current.isValid()) return;
+
+                     CHECK(current.model() == &instance_model_);
+                     connection_artifacts_->selected_instance_ =
+                         current.data(Qt::UserRole).value<Instance>();
+                     emit InstanceSelected();
+                   });
+
+  QObject::connect(ui_->rememberCheckBox, &QCheckBox::toggled, this, [this](bool checked) {
+    if (checked) {
```

Comment:
The `rememberCheckBox` is disabled until a connection is established. So only after a successful connection, the user can choose to remember the instance. 
When the instance disconnects (because the user did so) then this is not remembered anymore.

Let me know if that makes sense, or if another way makes more sense

Reply:
I read your other comment later. And I agree, lets leave it as is for now